src/coreclr/jit/importer.cpp

```diff
@@ -4387,6 +4387,12 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
         ssize_t offset;
         if (hasConstAddr)
         {
+            if (!isSharedStatic)
+            {
+                // Change SharedStatic to SimpleStaticKnownAddress
+                assert(fieldKind == FieldSeq::FieldKind::SimpleStatic);
+                fieldKind = FieldSeq::FieldKind::SimpleStaticKnownAddress;
+            }
```

Comment:
Nit: `hasConstAddr` -> `hasKnownAddr` for consistency.

There is no need to check for `!isSharedStatic`, since a constant can only refer to one location by definition.